### PR TITLE
Range helpers: Honor param_name when inferring fs

### PR DESCRIPTION
**Pull request by Tarrasch**
Range* allows you to specify the name of the parameter that is the date
the range is about. Another orthogonal feature is that you don't need to
provide a bulk_complete for the most common file system use case.

However, when using both these features doesn't work. This patch fixes
that.

### CODE REVIEW

**Comment by Tarrasch** [comment]
@ulzha @joeshaw, can anyone of you review please? :)

File: luigi/tools/range.py [hunk 1]
```diff
@@ -516,7 +516,7 @@ def missing_datetimes(self, task_cls, finite_datetimes):
         except NotImplementedError:
             return infer_bulk_complete_from_fs(
                 finite_datetimes,
-                lambda d: task_cls(self.datetime_to_parameter(d)),
```
Comment (joeshaw) [inline luigi/tools/range.py]: this change removes the use of `task_cls` as an argument to `missing_datetimes()`.  So it seems to me that either this patch should remove that argument (but I'm not sure what side effects that might have... when is `self.of_cls` ever different from `task_cls`?), or that instead of dropping `task_cls` as an argument to `_instantiate_task_cls()` this should call `self._instantiate_task_cls(task_cls, self.datetime_to_parameter(d))` instead.

In other words, we either need `task_cls` or we don't, and if we don't we shouldn't pass it around and should just use `self.of_cls` instead.

**Comment by Tarrasch** [comment]
I made a pretty big change to correct for the flaw that you found @joeshaw. How does it look now? Please see the amendment to the commit message too. :)

Also ping @ulzha

**Comment by joeshaw** [comment]
LGTM :+1: but definitely could use another pair of eyes as I am still new to the codebase.
